lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -1846,6 +1894,24 @@ FVectorType FVectorType::getAllConstDroppedType() {
              false);
 }
 
+/// Return this type with any type aliases recursively removed from itself.
+FIRRTLBaseType FVectorType::getAnonymousType() {
+  auto *impl = getImpl();
+
+  if (impl->anonymousType)
+    return impl->anonymousType;
+
+  // If this type is already anonymous, return it and remember for next time.
+  if (impl->elementType.getRecursiveTypeProperties().containsTypeAlias)
```

Comment:
The not is missing on the check ?
```suggestion
  if (!impl->elementType.getRecursiveTypeProperties().containsTypeAlias)
```

Reply:
Also same comment, maybe check this and early return, without setting the cached type.

Reply:
> The not is missing on the check ?

Totally, great catch. Thanks!